test/mpi_runtests.jl

```diff
@@ -258,22 +268,27 @@ end
     end
 
     @testset "Same seed gives same results" begin
-        # Note: all_to_all will give different results here, as it sorts slightly differently.
-        # The entries in the vectors are the same (tested above), but they appear in a
-        # different order.
-        add = BoseFS2C((1, 1, 1, 1, 1), (1, 0, 0, 0, 0))
-        H = BoseHubbardReal1D2C(add; v=2)
-        dv = DVec(add => 1)
-
-        dv_ptp = MPIData(copy(dv); setup=RMPI.mpi_point_to_point)
-        mpi_seed_CRNGs!(17)
-        df_ptp = lomc!(H, dv_ptp).df
-
-        dv_os = MPIData(copy(dv); setup=RMPI.mpi_one_sided, capacity=1000)
-        mpi_seed_CRNGs!(17)
-        df_os = lomc!(H, dv_os).df
-
-        @test df_ptp == df_os
+        for kwargs in (
+            (; setup=RMPI.mpi_point_to_point),
+            (; setup=RMPI.mpi_one_sided, capacity=10_000),
+            (; setup=RMPI.mpi_all_to_all),
+        )
+            # The entries in the vectors are the same (tested above), but they appear in a
+            # different order.
+            add = BoseFS2C((1, 1, 1, 1, 1), (1, 0, 0, 0, 0))
+            H = BoseHubbardReal1D2C(add; v=2)
+            dv = DVec(add => 1)
+
+            dv_1 = MPIData(copy(dv); kwargs...)
+            mpi_seed!(17)
+            df_1 = lomc!(H, dv_1; threading=false).df
+
+            dv_2 = MPIData(copy(dv); kwargs...)
+            mpi_seed!(17)
+            df_2 = lomc!(H, dv_2; threading=false).df
+
+            @test df_1 == df_1
```

Comment:
I assume this should be comparing the two?
```suggestion
            @test df_1 == df_2
```